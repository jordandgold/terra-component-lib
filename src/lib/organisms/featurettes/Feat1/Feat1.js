import React from "react";
import "./Feat1.scss";
import PropTypes from "prop-types";
import ButtonLink from "../../../ButtonLink/ButtonLink";

const Feat1 = ({ image, title, subtitle, text, cta, variant }) => {
  return (
    <section className={`ter-feat-one ter-feat-one--${variant}`}>
      <div
        className={`ter-feat-one__image-wrapper ter-feat-one__image-wrapper--${variant}`}
      >
        <img
          src={image.url}
          alt={image.altText}
          className="ter-feat-one__image"
        />
      </div>
      <div
        className={`ter-feat-one__content-container ter-feat-one__content-container--${variant}`}
      >
        {" "}
        <h3 className="ter-feat-one__title">{title}</h3>
        {subtitle && <h4 className="ter-feat-one__subtitle">{subtitle}</h4>}
        <p className="ter-feat-one__text">{text}</p>
        {cta && (
          <ButtonLink
            link={cta.link}
            text={cta.text}
            className="ter-button--primary--1"
          />
        )}
      </div>
    </section>
  );
};

export default Feat1;

Feat1.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  text: PropTypes.string.isRequired,
  cta: PropTypes.shape({
    link: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired
  }),
  variant: PropTypes.string.isRequired,
  image: PropTypes.shape({
    url: PropTypes.string.isRequired,
    altText: PropTypes.string.isRequired
  }).isRequired
};
