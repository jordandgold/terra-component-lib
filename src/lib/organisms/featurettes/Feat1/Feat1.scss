@import "../../../global-styles/terra.scss";

.ter-feat-one {
  padding: spacing("lg") spacing("md");

  @media (min-width: $breakpoint-xs) {
    max-width: 700px;
    margin: auto;
    padding-top: spacing("mega");
  }

  @media (min-width: $breakpoint-sm) {
    @include grid-container;
  }

  &--a {
    display: flex;
    flex-direction: column-reverse;

    @media (min-width: $breakpoint-sm) {
      display: flex;
      flex-direction: column-reverse;
    }

    @media (min-width: $breakpoint-md) {
      padding-bottom: 0;
      margin-bottom: 0;
      @include grid-container;
    }
  }

  &--b,
  &--c {
    @media (min-width: $breakpoint-sm) {
      padding: spacing("mega") spacing("md");
      align-items: center;
    }
  }

  &__image-wrapper {
    margin-bottom: spacing("md");

    @media (min-width: $breakpoint-sm) {
      @include grid-column(6);
    }

    &--a {
      margin-top: spacing("lg");
      margin-bottom: 0;

      @media (min-width: $breakpoint-md) {
        @include grid-column(6);
        align-self: self-end;
        margin-top: 0;
      }
    }

    &--b,
    &--c {
      @media (min-width: $breakpoint-sm) {
        margin-bottom: 0;
      }
    }
  }

  &__image {
    width: 100%;
  }

  &__content-container {
    @media (min-width: $breakpoint-sm) {
      @include grid-column(6);
    }

    &--a {
      @media (min-width: $breakpoint-md) {
        @include grid-column(6);
      }
    }
  }

  &__title {
    font-size: font-size("lg");
  }

  &__subtitle {
    font-size: font-size("base");
    font-weight: font-weight("light");
    color: grayscale("gray-7");
  }

  &__text {
    margin-top: spacing("md");
    margin-bottom: 0;
  }

  .ter-button {
    display: block;
    margin-top: spacing("md");

    @media (min-width: $breakpoint-xs) {
      display: inline-block;
      padding-right: spacing("lg");
      padding-left: spacing("lg");
      // max-width: 200px;
    }

    @media (min-width: $breakpoint-md) {
      margin-top: spacing("md");
    }
  }
}
