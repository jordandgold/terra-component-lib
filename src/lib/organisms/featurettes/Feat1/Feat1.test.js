import React from "react";
import { shallow } from "enzyme";
import Feat1 from "./Feat1";

describe("Feat1", () => {
  let wrapper;

  beforeEach(() => {
    const mockTitle = "Design is good.";
    const mockSubtitle = "You should know about it, maybe?";
    const mockText =
      "Cillum et laboris aliquip consequat. Consectetur commodo nisi laborum voluptate. Commodo est ullamco pariatur ut nostrud pariatur.";
    const mockCTA = { link: "#", text: "Button" };
    const mockImage = { url: "./1-to-1.png", altText: "placeholder" };
    wrapper = shallow(
      <Feat1
        title={mockTitle}
        subtitle={mockSubtitle}
        text={mockText}
        cta={mockCTA}
        variant="a"
        image={mockImage}
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
