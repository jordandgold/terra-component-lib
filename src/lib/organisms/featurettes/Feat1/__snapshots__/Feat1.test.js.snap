// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feat1 should match the snapshot 1`] = `
<section
  className="ter-feat-one ter-feat-one--a"
>
  <div
    className="ter-feat-one__image-wrapper ter-feat-one__image-wrapper--a"
  >
    <img
      alt="placeholder"
      className="ter-feat-one__image"
      src="./1-to-1.png"
    />
  </div>
  <div
    className="ter-feat-one__content-container ter-feat-one__content-container--a"
  >
     
    <h3
      className="ter-feat-one__title"
    >
      Design is good.
    </h3>
    <h4
      className="ter-feat-one__subtitle"
    >
      You should know about it, maybe?
    </h4>
    <p
      className="ter-feat-one__text"
    >
      Cillum et laboris aliquip consequat. Consectetur commodo nisi laborum voluptate. Commodo est ullamco pariatur ut nostrud pariatur.
    </p>
    <ButtonLink
      className="ter-button--primary--1"
      link="#"
      text="Button"
    />
  </div>
</section>
`;
